coderd/database/queries/dbcrypt.sql

```diff
@@ -0,0 +1,18 @@
+-- name: GetDBCryptKeys :many
+SELECT * FROM dbcrypt_keys ORDER BY number ASC;
+
+-- name: RevokeDBCryptKey :exec
+UPDATE dbcrypt_keys
+SET
+	revoked_key_digest = active_key_digest,
+	active_key_digest = revoked_key_digest,
+	revoked_at = CURRENT_TIMESTAMP
```

Comment:
Ok, this is the part where I got lost.

Table `dbcrypt_keys` stores encryption keys, so I would expect to keep one digest per record, but we store revoked and active values, and exchange them here. Most likely it's me being ignorant, but what is the purpose of this construct?

Reply:
My assumption is that a security person would like to have a record of when a key was revoked. Keeping the `revoked_key_digest` around for reference (along with `revoked_at`) lets them see that fairly simply by inspecting the table.

Reply:
Wouldn't it be simpler if we just keep adding new DBCryptKeys, and assume that the last one is active? I mean, adding a new key revokes the previous one.

Reply:
Adding a new key doesn't immediately revoke previous keys - key revocation is done via a CLI command (second PR). I wanted to intentionally decouple the process of adding a new key from the process of re-encrypting data.

Reply:
All clear now, thanks!